libsolidity/formal/SMTEncoder.cpp

```diff
@@ -449,11 +449,15 @@ void SMTEncoder::endVisit(UnaryOperation const& _op)
 	if (_op.annotation().type->category() == Type::Category::RationalNumber)
 		return;
 
-	if (TokenTraits::isBitOp(_op.getOperator()))
+	if (TokenTraits::isBitOp(_op.getOperator()) && !*_op.annotation().userDefinedFunction)
 		return bitwiseNotOperation(_op);
 
 	createExpr(_op);
 
+	// User-defined operators are essentially function calls.
+	if (*_op.annotation().userDefinedFunction)
+		return;
```

Comment:
@leonardoalt Is this the right way to handle user-defined operators? I suspect not since my test are not detecting assertion violations as they do when I call operator functions directly, but `createExpr(Expression)` is basically what seems to happen for an internal function call.

Reply:
We can always just emit a warning about user-defined operators not being supported by the smtchecker yet and take it from there. Not sure it's a good idea to try to properly implement them there in this PR.

Reply:
Agree with @ekpyron

Reply:
> but `createExpr(Expression)` is basically what seems to happen for an internal function call.

This is only in the SMTEncoder side though. BMC and CHC handle control flow separately, so maybe some extra support is needed there to interpret them as function calls.

Reply:
Warning added:

```
User-defined operators are not yet supported by SMTChecker. This invocation of operator + has been ignored, which may lead to incorrect results.
```

Other than that I left the things I added as is - since there wasn't that much of it anyway. Just a few `if`s preventing user-defined operators being processed as normal operators even with the warning. Do you think it can stay that way or should I actually get rid of the `createExpr(_op)` part when it's a custom operator?